Add `.timer` & various other systemd extensions (fixes #7161)

vorburger opened the pull request:
## Description

Fixes #7161.

## Checklist:

- [X] **I am adding a new extension to a language.**
  - TODO [ ] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      <!-- Replace FOOBAR with the new extension, and KEYWORDS with keywords unique to the language. Repeat for each extension added. -->
      - https://github.com/search?type=code&q=NOT+is%3Afork+path%3A*.FOOBAR+KEYWORDS
  - TODO [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [X] I have verified that there is no need to include a change to the heuristics to distinguish my language from others using the same extension (because all these new systemd specific file name extensions are not yet used by other other language).

lildude commented on lib/linguist/languages.yml:
```diff
@@ -8468,10 +8468,28 @@ crontab:
   language_id: 705203557
 desktop:
   type: data
+  # https://en.wikipedia.org/wiki/INI_file
```
These are all in the wrong place. If they're INI files, we already have a [section](https://github.com/github-linguist/linguist/blob/1a61b82c42754ff84f1a3e108150a2a7461c10bc/lib/linguist/languages.yml#L3056) for that.

vorburger replied:
> These are all in the wrong place. If they're INI files, we already have a [section](https://github.com/github-linguist/linguist/blob/1a61b82c42754ff84f1a3e108150a2a7461c10bc/lib/linguist/languages.yml#L3056) for that.

But then the already existing (systemd) ".service" is in the wrong place as well? Do you want me to move that one too? Otherwise it's inconsistent. Full disclaimer: I don't yet fully understand what difference "desktop" (which are also INI) VS "ini" language makes either way. What I could do is raise a separate small PR, just to first move . service?

lildude replied:
🤔 good point. Leave these here. I'd forgotten @Alhadis added `.service` in https://github.com/github-linguist/linguist/pull/5768 and makes the point the files are based on desktop which is essentially INI.

We def don't need the comments and only really want files and extensions which meet our usage requirements, along with a sample of each.

Alhadis replied:
@lildude Yeah, something I didn't address in #5768 was the observation that the "Desktop" language ought to be rolled into INI (which appears to be the correct course of action, AFAICT), because apart from having distinct syntax highlighting grammars, each entry refers to essentially the same format.

I only added `.service` to Desktop's file extensions because its usage appeared to be affiliated with <samp>systemd(1)</samp> and I didn't have time to look into whether or not it was worth merging Desktop into INI.

lildude reviewed (commented):
> How strictly does this project require "including a real-world usage sample for all extensions added in this PR"? I sure could copy/paste a random example from those "Search results for each extension" linked above, but... what's the point of that? (I'm happy to do so, if it's required to merge this, of course; just wasn't sure if really was a hard-requirement.)

It is. They're used to train the classifier.

Also every filename and extension needs to meet our usage requirements. One lesser used file or extension will hold up the PR until it meets the usage requirements.

vorburger commented:
> > How strictly does this project require "including a real-world usage sample for all extensions added in this PR"? I sure could copy/paste a random example from those "Search results for each extension" linked above, but... what's the point of that? (I'm happy to do so, if it's required to merge this, of course; just wasn't sure if really was a hard-requirement.)
> 
> It is. They're used to train the classifier.

ACK, thanks; I'll add samples.

> Also every filename and extension needs to meet our usage requirements. One lesser used file or extension will hold up the PR until it meets the usage requirements.

Do you prefer separate small PRs - one for each?

lildude commented:
> Do you prefer separate small PRs - one for each?

One PR is fine, and be sure to only include extensions that meet our usage requirements as the PR will be held up waiting for any that don't meet our requirements.

vorburger commented:
> We def don't need the comments

Inline comments now removed.

> and only really want files and extensions which meet our usage requirements

These do, do the best of my understanding of your criteria, and knowledge. If I misunderstood, let me know, and I'll remove.

See https://github.com/github-linguist/linguist/issues/7161#issuecomment-2543120512 for which extensions I have opted NOT to include.

> along with a sample of each

Samples now added.

lildude reviewed (changes requested):
Please check the usage for each of these new files and extensions and add searches for the filenames.

A quick look at several of your results indicates a lot of these don't meet our usage requirements as repos can reasonably be expected to have more than one of them, which means the number of files criteria needs to be met. Some of the filenames you're adding only have single digit results.

Please address the test failures.

> I have included a real-world usage sample for all extensions added in this PR:
> 
>     * Sample source(s):
>       * [URL to each sample source, if applicable]
>     * Sample license(s):

This needs to be filled in for every sample.
